### PR TITLE
Issue #3411006: Topic and Group type taxonomy permission missing

### PR DESCRIPTION
**Pull request by vcsvinicius**
## Problem
When the Taxonomy Access Fix module was enabled, the Topic and Group type don't be displayed as expected and this fields is mandatory to create new entities.

## Solution
Created a new configuration modify to add this permission when the Taxonomy Access Fix module are enable.

## Issue tracker
[PROD-27790](https://getopensocial.atlassian.net/browse/PROD-27790)
[#3411006](https://www.drupal.org/project/social/issues/3411006)

## Theme issue tracker
N/A

## How to test
- [ ] Install and enable Taxonomy Access Fix
- [ ] Enable Social Flexible Group
- [ ] Clear cache
- [ ] Try to create a Topic and Group
- [ ] I should be able to see the Type fields and be able to fill them

### Before merge
- [ ] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [ ] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [ ] New features or changes to existing features are covered by tests, either unit (preferably) or behat
- [ ] Update path is tested. New hook_updates should respect update order, right naming convention and consider hook_post_update code
- [ ] Module can be safely uninstalled. Update/implement hook_uninstall and make sure that removed configuration or dependencies are removed/uninstalled
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [ ] New UI elements, or changes on UI elements are approved by the design team
- [ ] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

## Screenshots
N/A

## Release notes
N/A

## Change Record
N/A

## Translations
N/A

[PROD-27790]: https://getopensocial.atlassian.net/browse/PROD-27790?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3665-pprcrhepn2ed2xb1fbyyg1llc10i5ecq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/658c14aded5de7d370a2111a

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3665-pprcrhepn2ed2xb1fbyyg1llc10i5ecq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/658c14aded5de7d370a2111a

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3665-pprcrhepn2ed2xb1fbyyg1llc10i5ecq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/658c14aded5de7d370a2111a

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3665-pprcrhepn2ed2xb1fbyyg1llc10i5ecq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/658c14aded5de7d370a2111a

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3665-pprcrhepn2ed2xb1fbyyg1llc10i5ecq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/658c14aded5de7d370a2111a

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3665-pprcrhepn2ed2xb1fbyyg1llc10i5ecq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/658c14aded5de7d370a2111a

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3665-pprcrhepn2ed2xb1fbyyg1llc10i5ecq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/658c14aded5de7d370a2111a

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3665-pprcrhepn2ed2xb1fbyyg1llc10i5ecq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/658c14aded5de7d370a2111a

**Comment by ronaldtebrake** [comment]
I think we still need a update hook, to make sure that existing installs, that have taxonomy_access_fix already enabled or where it's being enabled, also get an update of these permissions. This does seem to work nicely on install 👍

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3665-pprcrhepn2ed2xb1fbyyg1llc10i5ecq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/658c14aded5de7d370a2111a

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3665-pprcrhepn2ed2xb1fbyyg1llc10i5ecq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/658c14aded5de7d370a2111a

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3665-pprcrhepn2ed2xb1fbyyg1llc10i5ecq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/658c14aded5de7d370a2111a

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3665-pprcrhepn2ed2xb1fbyyg1llc10i5ecq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/658c14aded5de7d370a2111a

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3665-pprcrhepn2ed2xb1fbyyg1llc10i5ecq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/658c14aded5de7d370a2111a

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3665-pprcrhepn2ed2xb1fbyyg1llc10i5ecq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/658c14aded5de7d370a2111a

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3665-pprcrhepn2ed2xb1fbyyg1llc10i5ecq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/658c14aded5de7d370a2111a

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3665-pprcrhepn2ed2xb1fbyyg1llc10i5ecq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/658c14aded5de7d370a2111a

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3665-pprcrhepn2ed2xb1fbyyg1llc10i5ecq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/658c14aded5de7d370a2111a

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3665-pprcrhepn2ed2xb1fbyyg1llc10i5ecq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/658c14aded5de7d370a2111a

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3665-pprcrhepn2ed2xb1fbyyg1llc10i5ecq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/658c14aded5de7d370a2111a

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3665-pprcrhepn2ed2xb1fbyyg1llc10i5ecq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/658c14aded5de7d370a2111a

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3665-pprcrhepn2ed2xb1fbyyg1llc10i5ecq.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/658c14aded5de7d370a2111a

File: modules/social_features/social_topic/config/modify/social_topic.authenticated_taxonomy_permissions.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+dependencies:
+  module:
+    - taxonomy_access_fix
```
Comment (navneet0693) [inline modules/social_features/social_topic/config/modify/social_topic.authenticated_taxonomy_permissions.yml:3]: Does this means that config modify will check for existence of module first? Or, just add these properties to current configuration?

File: modules/social_features/social_topic/social_topic.install [hunk 2]
```diff
@@ -563,3 +563,33 @@ function social_topic_update_120001(): string {
   // Output logged messages to related channel of update execution.
   return $updater->logger()->output();
 }
+
+/**
+ * Add taxonomy permission with the Taxonomy Access Fix is enabled.
+ */
+function social_topic_update_120002(): void {
+  // Nothing to do if the module is not installed.
+  if (!\Drupal::moduleHandler()->moduleExists("taxonomy_access_fix")) {
```
Comment (navneet0693) [inline modules/social_features/social_topic/social_topic.install:572]: Just a minor improvement for single quotes.

File: modules/social_features/social_topic/social_topic.install [hunk 3]
```diff
@@ -563,3 +563,33 @@ function social_topic_update_120001(): string {
...
+  // Load the permission.
+  $entity_type_manager = \Drupal::entityTypeManager();
+  /** @var \Drupal\user\RoleInterface $role */
+  $role = $entity_type_manager->getStorage('user_role')->load('authenticated');
+
+  // If the role is not have the permission, grant permission.
+  foreach ($taxonomy_permissions as $taxonomy_permission) {
+    if (!$role->hasPermission($taxonomy_permission)) {
+      $role->grantPermission($taxonomy_permission);
+      $role->save();
```
Comment (navneet0693) [inline modules/social_features/social_topic/social_topic.install:592]: I think $role->save() should be called outside foreach loop.

**Review by navneet0693 — CHANGES_REQUESTED** [review]

File: modules/social_features/social_topic/config/modify/social_topic.authenticated_taxonomy_permissions.yml (hunk 1 above)
Comment (vcsvinicius) [reply modules/social_features/social_topic/config/modify/social_topic.authenticated_taxonomy_permissions.yml:3]: @navneet0693 This means the permission only will be allowed if the Taxonomy Access Fix module is enabled.

Comment (navneet0693) [reply modules/social_features/social_topic/config/modify/social_topic.authenticated_taxonomy_permissions.yml:3]: @vcsvinicius Thank you for educating me!

**Review by navneet0693 — APPROVED** [review]

**Comment by navneet0693** [comment]
@vcsvinicius I tried re-running the failed tests, but I think you might need to rebase the PR. It's good to go!

**Comment by navneet0693** [comment]
Thank you for rebasing the PR @vcsvinicius !

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3665-qc2ydcjlkzvpu7ggcynaqhqhv28bjxkg.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/65b393b89cfb585b14e640dd

**Comment by vcsvinicius** [comment]
I cherry-picked to 12.0.x and 12.1.x branch

**Pull request merged**